src/ceph-volume/ceph_volume/devices/lvm/deactivate.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import logging
+import sys
+from textwrap import dedent
+from ceph_volume import conf
+from ceph_volume.util import encryption, system
+from ceph_volume.api.lvm import get_lvs_by_tag
+
+logger = logging.getLogger(__name__)
+
+
+def deactivate_osd(osd_id=None, osd_uuid=None):
+
+    lvs = []
+    if osd_uuid:
+        lvs = get_lvs_by_tag('ceph.osd_fsid={}'.format(osd_uuid))
+    else:
+        lvs = get_lvs_by_tag('ceph.osd_id={}'.format(osd_id))
+
+    try:
+        data_lv = next(lv for lv in lvs if lv.tags['ceph.type'] in ['data', 'block'])
+    except StopIteration:
+        logger.error('No data or block LV found for OSD {}'.format(osd_id))
+        sys.exit(1)
+    conf.cluster = data_lv.tags['ceph.cluster_name']
+    logger.debug('Found cluster name {}'.format(conf.cluster))
+
+    tmpfs_path = '/var/lib/ceph/osd/{}-{}'.format(conf.cluster, osd_id)
+    logger.info('Unmounting osd-{}\'s tmpfs path at {}'.format(
+        osd_id,
+        tmpfs_path))
+    system.unmount(tmpfs_path)
```

Comment:
i think this might not be tmpfs... so this should probalby only try to unmount if it is?

Reply:
Yes though the code won't get this far if the OSD is not LVM based (https://github.com/ceph/ceph/pull/32179/files#diff-d5b0389680f14e4c45e3f8ccafa7707aR21). But some extra fencing is probably a good thing.

Reply:
fwiw currently inside the ceph-daemon containers there is no tmpfs, and it's lv based

Reply:
got it, I'll revisit tomorrow.

Reply:
@liewegas Does this work for you?